core/valor_core/geometry.py

```diff
@@ -0,0 +1,1440 @@
+import io
+import json
+from base64 import b64decode, b64encode
+from dataclasses import dataclass
+from typing import Any, Optional, TypedDict, Union
+
+import numpy as np
+import PIL.Image
+
+
+def calculate_bbox_intersection(bbox1: np.ndarray, bbox2: np.ndarray) -> float:
```

Comment:
note - This works for axis-aligned bounding boxes but we lose support for anything transformed.